man/systemd.network.xml

```diff
@@ -991,6 +991,13 @@
             Supported only on IPv6. Defaults to false.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>IPv4DuplicateAddressDetection=</varname></term>
+          <listitem>
+            <para>Takes a boolean. When true performs IPv4 Duplicate Address Detection. See
+            <ulink url="https://tools.ietf.org/html/rfc5227">RFC 5224</ulink>. Defaults to false.</para>
```

Comment:
No, let's not add yet another option. If users wants DAD, then most likely they want both ipv4 and ipv6, and don't care about internal implementation details, i.e. whether this is the kernel that is doing this or the user space.

Let's do the following: make DuplicateAddressDetection take: yes|no, "both", "ipv4", "ipv6". If yes|no, then warn about the inverted meaning and set according to current semantics. If "ipv4", enable for IPv4, if "ipv6", enable for IPv6, if "both" enable for both.

This is easier for the users, and will give us more flexibility to add new variants of this setting in the future.